src/interface/AbstractPortal.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+//TODO: Add hooks with basic implementation
+abstract contract AbstractPortal {
+  function attest(
+    bytes32 schemaId,
```

Comment:
Should the schemaId be part of the `validationPayload`?  Many implementations may want to record the schema id in state rather than passing it with every transaction